### PR TITLE
Api structural variant rebased

### PR DESCRIPTION
**Pull request by sandertan**
Rebased version of https://github.com/thehyve/cbioportal/pull/24

### CODE REVIEW

**Comment by dionnezaal** [comment]
Have added second sample to test database to make the MyBatisRepository test `fetchStructuralVariantsWithSampleIdentifiers` indeed check if the sampleId is filtered.

File: web/src/main/java/org/cbioportal/web/StructuralVariantController.java
Comment (rnugraha) [inline web/src/main/java/org/cbioportal/web/StructuralVariantController.java]: Why using empty array? Shouldn't this use structuralVariantFilter.getMolecularProfileIds()?

File: web/src/test/java/org/cbioportal/web/StructuralVariantControllerTest.java
Comment (rnugraha) [inline web/src/test/java/org/cbioportal/web/StructuralVariantControllerTest.java]: Please provide test with the case where filter will be `molecularProfileIds` and `hugeGeneSymbols`. Even better provide failing test.

File: web/src/main/java/org/cbioportal/web/StructuralVariantController.java
Comment (dionnezaal) [reply web/src/main/java/org/cbioportal/web/StructuralVariantController.java]: You are correct! This is a mistake, should indeed be structuralVariantFilter.getMolecularProfileIds().

**Comment by dionnezaal** [comment]
@rnugraha Have fixed mistake in controller with molecularProfileId. Also added two more test to the controller, one for when molecularProfileIds are given as input, and failing test when input contains both MolecularProfileId and SampleMolecularIdentifiers.

**Comment by rnugraha** [comment]
@dionnezaal Checks are still failing, could you fix them? Maybe this is also due to the root branch (Sander's)

@sandertan @pieterlukasse I would really appreciate merging this PR ASAP. Do we have any concerns that hold this PR to be merged?

**Comment by pieterlukasse** [comment]
@rnugraha It will have to be reviewed before it gets merged. Why did we move away from the PR where review had started? Just checking...I can continue review here.

**Comment by dionnezaal** [comment]
@rnugraha Will work on it as soon as possible

@pieterlukasse This version was rebased by sander from https://github.com/thehyve/cbioportal/pull/24 when I had a day off, afterwards continued on this rebased version.

File: model/src/main/java/org/cbioportal/model/StructuralVariant.java
Comment (rnugraha) [inline model/src/main/java/org/cbioportal/model/StructuralVariant.java]: 2018

File: persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/StructuralVariantMyBatisRepository.java
Comment (rnugraha) [inline persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/StructuralVariantMyBatisRepository.java]: Add copyright

File: persistence/persistence-mybatis/src/main/resources/org/cbioportal/persistence/mybatis/StructuralVariantMapper.xml
Comment (rnugraha) [inline persistence/persistence-mybatis/src/main/resources/org/cbioportal/persistence/mybatis/StructuralVariantMapper.xml]: Add copyright

File: persistence/persistence-mybatis/src/test/java/org/cbioportal/persistence/mybatis/StructuralVariantMyBatisRepositoryTest.java
Comment (rnugraha) [inline persistence/persistence-mybatis/src/test/java/org/cbioportal/persistence/mybatis/StructuralVariantMyBatisRepositoryTest.java]: Add copyright

File: service/src/main/java/org/cbioportal/service/StructuralVariantService.java
Comment (rnugraha) [inline service/src/main/java/org/cbioportal/service/StructuralVariantService.java]: Add copyright

File: service/src/main/java/org/cbioportal/service/StructuralVariantService.java
Comment (rnugraha) [inline service/src/main/java/org/cbioportal/service/StructuralVariantService.java:34]: Add new line

File: service/src/main/java/org/cbioportal/service/impl/StructuralVariantServiceImpl.java
Comment (rnugraha) [inline service/src/main/java/org/cbioportal/service/impl/StructuralVariantServiceImpl.java]: Add copyright

File: service/src/test/java/org/cbioportal/service/impl/StructuralVariantServiceImplTest.java
Comment (rnugraha) [inline service/src/test/java/org/cbioportal/service/impl/StructuralVariantServiceImplTest.java]: Add copyright

File: web/src/main/java/org/cbioportal/web/parameter/StructuralVariantFilter.java
Comment (rnugraha) [inline web/src/main/java/org/cbioportal/web/parameter/StructuralVariantFilter.java]: Add copyright

File: web/src/main/java/org/cbioportal/web/parameter/StructuralVariantFilter.java
Comment (rnugraha) [inline web/src/main/java/org/cbioportal/web/parameter/StructuralVariantFilter.java]: Add new line

File: web/src/test/java/org/cbioportal/web/StructuralVariantControllerTest.java
Comment (rnugraha) [inline web/src/test/java/org/cbioportal/web/StructuralVariantControllerTest.java]: Add copyright

File: persistence/persistence-mybatis/src/test/java/org/cbioportal/persistence/mybatis/StructuralVariantMyBatisRepositoryTest.java
Comment (rnugraha) [inline persistence/persistence-mybatis/src/test/java/org/cbioportal/persistence/mybatis/StructuralVariantMyBatisRepositoryTest.java]: Why comparing each attributes one by one? Can't we just assert the whole object?

**Review by rnugraha — CHANGES_REQUESTED** [review]

File: persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/StructuralVariantMyBatisRepository.java
Comment (rnugraha) [inline persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/StructuralVariantMyBatisRepository.java]: structuralVariantmapper pls changed to structuralVariantMapper. Pls maintain the camel case code styling.

File: persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/StructuralVariantMyBatisRepository.java
Comment (rnugraha) [inline persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/StructuralVariantMyBatisRepository.java]: structuralVariantmapper pls changed to structuralVariantMapper. Pls maintain the camel case code styling.

**Review by rnugraha — CHANGES_REQUESTED** [review]

**Review by oplantalech — APPROVED** [review]
Good job, I've revised all the code and it looks good.

**Comment by pieterlukasse** [comment]
@oplantalech please squash the commits.

**Comment by oplantalech** [comment]
@pieterlukasse Done

**Review by rnugraha — APPROVED** [review]

**Pull request merged**
